### PR TITLE
Main layout page for FoR

### LINKED ISSUE #4: Create main layout (top, left, right) page for FOR

### PR DESCRIPTION
**Pull request by netr0m**
This PR includes the updated UI for FoR (more or less) as per the design in Figma.
Resolves #4 
Added:
- `forecast-of-response` custom plugin
- `App` "package" in `forecast-of-response/src/components/App` [which contains all components](https://github.com/equinor/forecast-of-response-dmt/tree/feature/main-layout-page/web/custom-plugins/forecast-of-response/src/components/App) related to the FoR app itself (UI)
- Pre-commit spec
- TypeScript config
- Prettier config
- Jest config

Removed:
- Deleted previous `for-operations` custom-plugin. Replaced by `forecast-of-response` custom-plugin

### CODE REVIEW

File: web/custom-plugins/for-operations/src/Enums.ts [hunk 1]
```diff
@@ -1,11 +0,0 @@
-export enum OperationStatus {
-  UPCOMING = 'Upcoming',
-  IN_PROGRESS = 'In progress',
```
Comment (soofstad) [inline web/custom-plugins/for-operations/src/Enums.ts:3]: Not sure we need to delete the old for-operation stuff.
They could be used as a basis for future development

File: web/custom-plugins/forecast-of-response/src/Types.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  hidden: boolean
+  visibleDataSources: any
+  type: string
+  description: string
+  packages: any
+  models: any
+  actions: any
+  file_loc: string
+  data_source_aliases: any
+}
```
Comment (soofstad) [inline web/custom-plugins/forecast-of-response/src/Types.ts:19]: Good stuff! But this is DMT specific. And should be moved to the DMT package and imported from there.... in some way.

File: web/custom-plugins/forecast-of-response/src/Types.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+export type TApp = {
+  applications: any
+  settings: TSettings
+}
+
+export type TAppHeader = {
+  appName: string
+}
```
Comment (soofstad) [inline web/custom-plugins/forecast-of-response/src/Types.ts:28]: Interfaces which takes only one simple type as argument can have it's type decleration directly in the function declaration

File: web/custom-plugins/forecast-of-response/src/components/App/AppHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const Icons = styled.div`
+  display: flex;
+  align-items: center;
+  flex-direction: row-reverse;
+  > * {
+    margin-left: 40px;
+  }
+`
+
+export default (props: TAppHeader): JSX.Element => {
```
Comment (soofstad) [inline web/custom-plugins/forecast-of-response/src/components/App/AppHeader.tsx:16]: Not a fan of everything being called "AppHeader", "AppContent" etc.
Makes me wonder if there are other more "important" 'Header' components.

**Review by soofstad — CHANGES_REQUESTED** [review]
Veldig bra! Full kontoll på EDS og TypeScript!
Noen mindre kommentarer. Ta stiling til de selv, og kommenter hva du gjør.
Er lov å si at det ikke fikses nå, men legg da gjerne inn en `// TODO:` til senere.

File: web/custom-plugins/for-operations/src/Enums.ts (hunk 1 above)
Comment (netr0m) [reply web/custom-plugins/for-operations/src/Enums.ts:3]: Yeah, agreed! Will revert those deletions

File: web/custom-plugins/forecast-of-response/src/Types.ts (hunk 2 above)
Comment (netr0m) [reply web/custom-plugins/forecast-of-response/src/Types.ts:19]: @soofstad How about adding it to the "types.tsx" in dmt @ web/plugins/core-plugins/src/types.tsx?

Comment (netr0m) [reply web/custom-plugins/forecast-of-response/src/Types.ts:20]: Added a TODO for now.

Comment (netr0m) [reply web/custom-plugins/forecast-of-response/src/Types.ts:20]: Added a PR in equinor/data-modelling-tool/pull/907

**Pull request merged**